### PR TITLE
feat(input): full width on low screen resolution

### LINKED ISSUE #8795: Make the filter inputs full width by default on low screen resolution (mobile)
It make more sense to me to have all the input taking a full width when we are on mobile view.

So instead of this being the default:

<details><summary>Before</summary>

Screen Shot 2023-04-03 at 21 00 22

</details> 

I suggest to have this:

<details><summary>After</summary>

Screen Shot 2023-04-03 at 21 04 26

</details> 

This is what I am willing to on my project. I may work on that but I would like to be sure about the idea first.

Thanks!

### PR DESCRIPTION
**Pull request by soullivaneuh**
# Summary

closes #8795

### CODE REVIEW

**Comment by soullivaneuh** [comment]
I am not sure about what fix to provide for the [typecheck issue](https://github.com/marmelab/react-admin/actions/runs/4602030405/jobs/8130569948?pr=8797).

May you give me a hint?

**Comment by slax57** [comment]
> I am not sure about what fix to provide for the [typecheck issue](https://github.com/marmelab/react-admin/actions/runs/4602030405/jobs/8130569948?pr=8797).
> 
> May you give me a hint?
> 
> I followed that documentation: https://mui.com/material-ui/customization/theme-components/#theme-style-overrides

It's odd, but it seems that, although this syntax is documented, MUI is not willing to support it in the long term and hence not willing to support it in the TS types (source: https://github.com/mui/material-ui/issues/29319)

You might have to find another way to do this...

**Comment by soullivaneuh** [comment]
@slax57 Thanks!

That is quite a bad news because the other found I started to get into was to add the `fullWidth` property logic in each input component, which will lead to a maintenance nightmare in my opinion.

According to [that comment](https://github.com/mui/material-ui/issues/29319#issuecomment-952587677), we may be able to achieve our needs using `theme.components[Component].variants`. I'll dig onto that.

**Comment by soullivaneuh** [comment]
I got that fixed using variants! :tada: 

As the usage of `styleOverrides`  looks not recommended for the future, I also took the initiative to migrate the existing on a separated commit: b217bb621

File: packages/ra-ui-materialui/src/defaultTheme.ts [hunk 1]
```diff
@@ -22,22 +22,49 @@ export const defaultTheme = {
...
+                    props: {},
+                    style: {
                         backgroundColor: 'rgba(0, 0, 0, 0.04)',
+                        '&$disabled': {
+                            backgroundColor: 'rgba(0, 0, 0, 0.04)',
+                        },
                     },
                 },
-            },
+            ],
```
Comment (slax57) [inline packages/ra-ui-materialui/src/defaultTheme.ts:49]: Can you explain why you needed to change these lines too?

Comment (soullivaneuh) [reply packages/ra-ui-materialui/src/defaultTheme.ts:49]: Please see: https://github.com/marmelab/react-admin/pull/8797#issuecomment-1496479298

> As the usage of styleOverrides looks not recommended for the future, I also took the initiative to migrate the existing on a separated commit: https://github.com/marmelab/react-admin/commit/b217bb621e94097bcfe38a58ee75c26585092353

If you prefer to keep the change to the target, I can simply drop this commit. :+1:

**Comment by fzaninotto** [comment]
This PR doesn't fix the aforementioned problem, see the deployed storybook https://react-admin-storybook-jxzza3gyv-marmelab.vercel.app/?path=/story/ra-ui-materialui-list-list--filters

image

**Comment by fzaninotto** [comment]
No news for some time, closing.

**Comment by fzaninotto** [comment]
Superseded by #9169

**Pull request closed without merging**